### DISCUSSION TITLE
Sktime Forecasting GridSerachCV  :  userwarning

### DISCUSSION THREAD

**Discussion by Ajay-user** [alan-turing-institute/sktime#3498] [General]
Hi , I'm new to sktime

I'm getting this user waring when using forecastingGridSearchCV 
**UserWarning**: y_pred and y_true do not have the same column index. This may indicate incorrect objects passed to the metric. Indices of y_true will be used for y_pred.

Is there any work-around ? or is it normal?

[sktime documentation link](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html
)

```python
from sktime.datasets import load_shampoo_sales
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.model_selection import ForecastingGridSearchCV, ExpandingWindowSplitter

y = load_shampoo_sales()

fh = [1,2,3]

cv = ExpandingWindowSplitter(
    start_with_window=True,
    fh=fh)

forecaster = NaiveForecaster()

param_grid = {"strategy" : ["last", "mean", "drift"]}

gscv = ForecastingGridSearchCV(
    forecaster=forecaster,
    param_grid=param_grid,
    cv=cv)

gscv.fit(y)
```

_UserWarning_: _y_pred and y_true do not have the same column index. This may indicate incorrect objects passed to the metric. Indices of y_true will be used for y_pred._

**Comment by fkiraly** [comment]
@Ajay-user, this does not sound right. Thanks for reporting!

`sktime` probably coerces your containers internally to the right ones, but it sounds like there might be a bug hidden, where one forecaster or transformer produces column names that do not agree with those in training.

Looks like a bug, will open a report.

**Reply by fkiraly** [reply]
here: https://github.com/sktime/sktime/issues/3510

**Comment by Ajay-user** [comment]
UserWarning: NotImplementedWarning:

 NaiveForecaster does not have a custom `update` method implemented. NaiveForecaster will be refit each time `update` is called with update_params=True.

In Sktime NaiveForecaster documentation under the methods I can find `update_predict` method, but I get this warning

[NaiveForecaster documentation](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.naive.NaiveForecaster.html)


code : 

`

from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.datasets import load_airline

y = load_airline()

y_train, y_test = temporal_train_test_split(y, fh=np.arange(1,37))



from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.model_selection import SlidingWindowSplitter


cv = SlidingWindowSplitter(fh=1,)

window = pd.Series(y_train.iloc[-10:], index=y_train.index[-10:])

test_data = pd.concat([window, y_test])

forecaster = NaiveForecaster(strategy='last' ,window_length=12, sp=12).fit(y_train)

y_pred = forecaster.update_predict(test_data, cv)

plot_series(y_train, y_test ,y_pred)

`
